projects/driver_controls/test/test_ads1015.c

```diff
@@ -0,0 +1,210 @@
+  // Tests for storage being a null pointer.
+  TEST_ASSERT_EQUAL(
+      STATUS_CODE_INVALID_ARGS,
+      ads1015_configure_channel(NULL, ADS1015_CHANNEL_1, true, prv_callback, &storage));
+}
+
+void test_ads1015_read_invalid_input(void) {
+  int16_t reading;
+  GPIOAddress ready_pin = {
+    .port = GPIO_PORT_B,  //
+    .pin = 2,             //
+  };
+  Ads1015Storage storage;
+  ads1015_init(&storage, I2C_PORT_2, ADS1015_ADDRESS_GND, &ready_pin);
+  ads1015_configure_channel(&storage, ADS1015_CHANNEL_0, true, NULL, &storage);
+  ads1015_configure_channel(&storage, ADS1015_CHANNEL_1, true, NULL, &storage);
+  ads1015_configure_channel(&storage, ADS1015_CHANNEL_2, true, NULL, &storage);
+  ads1015_configure_channel(&storage, ADS1015_CHANNEL_3, true, NULL, &storage);
+  // Tests a correct use of the function.
+  TEST_ASSERT_EQUAL(STATUS_CODE_OK, ads1015_read_converted(&storage, ADS1015_CHANNEL_0, &reading));
+  TEST_ASSERT_EQUAL(STATUS_CODE_OK, ads1015_read_raw(&storage, ADS1015_CHANNEL_2, &reading));
+  // Tests for storage being null.
+  TEST_ASSERT_EQUAL(STATUS_CODE_INVALID_ARGS,
+                    ads1015_read_converted(NULL, ADS1015_CHANNEL_2, &reading));
+  TEST_ASSERT_EQUAL(STATUS_CODE_INVALID_ARGS, ads1015_read_raw(NULL, ADS1015_CHANNEL_2, &reading));
+  // Tests for reading being null.
+  TEST_ASSERT_EQUAL(STATUS_CODE_INVALID_ARGS,
+                    ads1015_read_converted(&storage, ADS1015_CHANNEL_2, NULL));
+  TEST_ASSERT_EQUAL(STATUS_CODE_INVALID_ARGS, ads1015_read_raw(&storage, ADS1015_CHANNEL_2, NULL));
+
+  // Tests for out of bound channel.
+  TEST_ASSERT_EQUAL(STATUS_CODE_INVALID_ARGS,
+                    ads1015_read_converted(&storage, NUM_ADS1015_CHANNELS, &reading));
+  TEST_ASSERT_EQUAL(STATUS_CODE_INVALID_ARGS,
+                    ads1015_read_raw(&storage, NUM_ADS1015_CHANNELS, &reading));
+}
+
+// This test checks if the callbacks are called properly for enabled channels.
+void test_ads1015_channel_callback(void) {
+  GPIOAddress ready_pin = {
+    .port = GPIO_PORT_B,  //
+    .pin = 2,             //
+  };
+  Ads1015Storage storage;
+  bool callback_called_0 = false;
+  bool callback_called_1 = false;
+  bool callback_called_2 = false;
+  bool callback_called_3 = false;
+  ads1015_init(&storage, I2C_PORT_2, ADS1015_ADDRESS_GND, &ready_pin);
+  ads1015_configure_channel(&storage, ADS1015_CHANNEL_0, true, prv_callback, &callback_called_0);
+  ads1015_configure_channel(&storage, ADS1015_CHANNEL_1, false, prv_callback, &callback_called_1);
+  ads1015_configure_channel(&storage, ADS1015_CHANNEL_2, true, prv_callback, &callback_called_2);
+  ads1015_configure_channel(&storage, ADS1015_CHANNEL_3, false, prv_callback, &callback_called_3);
+  delay_ms(50);
+  TEST_ASSERT_EQUAL(true, callback_called_0);
+  TEST_ASSERT_EQUAL(false, callback_called_1);
+  TEST_ASSERT_EQUAL(true, callback_called_2);
+  TEST_ASSERT_EQUAL(false, callback_called_3);
+}
+
+// Tests a common order of enabling disabling channels.
```

Comment:
s/enabling disabling/enabling and disabling/ ?

Also, what's the difference between this test and all channels enabled? What case does this test that isn't covered by other tests or vice-versa?